To-do 리스트 리팩터링 하기

bbhye1 opened the pull request:
### **구현 목표 **

-  To-do 리스트를 redux를 이용해서 상태관리 하도록 변경하기
-  reducer들의 테스트 추가하기 

### **구현 조건**

- 모든 기능은 TDD로 구현합니다. 컴포넌트와 reducer에 대한 테스트가 존재해야 합니다. 먼저 테스트를 추가하고 commit 한 뒤 실제 기능을 추가하고 commit합니다. commit 메시지 로그는 아래와 같이 남아야 합니다.
- if는 사용할 수 있어도 else 사용하실 수 없습니다. GuardClauses 방법을 사용해주세요.
- switch는 사용하실 수 없습니다.
- let은 사용하실 수 없습니다. const만을 사용하여 문제를 해결해주세요.
- 함수 이름과 변수 이름에 줄임말은 사용하실 수 없습니다. 길더라도 명확하게 써주세요.
- indent(인덴트, 들여쓰기) depth를 1로 유지해주세요. 예를 들어, for문 안에 if문이 있으면 indent depth는 2입니다. depth를 줄이는 좋은 방법은 함수(또는 메소드)를 분리하면 됩니다.

### **구현 내용**
-TDD로 리덕스를 생성하는 부분부터 구현했습니다. 
- 뭔가 중간에 redux랑 App컴포넌트랑 연결하는 부분부터 순서가 조금 정리가 안되고 뭉터기로 지나간 것같아 약간 아쉽습니다 ㅠㅠ;; 
-App 컴포넌트 테스트가 coverage 100%달성이 안돼서 그 부분을 앞으로 수정해야겠습니다.

bbhye1 committed:
Refactoring App.jsx

상태를 관리해주는 부분과 화면을 그려주는 부분을 나눠주었다.

bbhye1 committed:
Install redux & react-redux.

리덕스와 리액트-리덕스를 설치했다.

bbhye1 committed:
App.js에 대한 테스트를 추가한다.

bbhye1 committed:
App.jsx에 대한 테스트 내용을 구현한다.

bbhye1 committed:
App.jsx을 리팩토링한다.

리덕스를 이용해 store를 생성해서 리덕스의 상태를 화면에 보여준다.

bbhye1 committed:
App.jsx를 리팩토링한다.

store를 따로 파일로 만들어 분리한다.

bbhye1 committed:
reducer에 대한 테스트를 생성한다.

bbhye1 committed:
reducer를 구현한다.

reducer를 빠르게 구현한다.

bbhye1 committed:
reducer를 리팩토링한다.

bbhye1 committed:
reducer.jsx에 테스트를 추가한다.

reducer.jsx에 addTask 테스트를 추가한다.

bbhye1 committed:
reducer를 구현한다.

reducer를 하드코딩해서 구현한다.

bbhye1 committed:
reducer를 리팩토링 한다.

reducer를 리팩토링할 때 필요한 테스트의 부분도 수정했다.

bbhye1 committed:
reducer에 대한 테스트를 추가한다.

addTask를 task title이 있을 때와 없을때의 문맥으로 나누어 테스트를
추가했다.

bbhye1 committed:
reducer를 구현한다.

bbhye1 committed:
reducer 테스트를 수정한다.

테스트에 id가 uhdefined되지 않도록 newId를 넣어준다.

bbhye1 committed:
reducer에 deleteTask 테스트를 추가한다.

bbhye1 committed:
reducer를 구현한다.

reducer를 하트코딜해서 구현한다.

bbhye1 committed:
reducer를 리팩토링한다.

reducer의 deleteTask를 리팩토링한다.

bbhye1 committed:
reducer.jsx를 리팩토링한다.

bbhye1 committed:
reducer.test.jsx를 리팩토링한다.

bbhye1 committed:
App.jsx를 리팩토링한다.

리덕스에 연결하고 action들과 dispatch를 생성한다.

hannut91 commented on src/reducer.test.jsx:
```diff
@@ -0,0 +1,85 @@
+    });
+
+    context('without task title', () => {
+      it("dosen't work", () => {
```
`task title`없이 액션을 실행하면 `동작하지 않는다`라는 의미가 무엇인지 조금 더 의도를 드러내면 좋을 것 같아요. 
사실 tasks의 길이가 늘어나지 않는걸 표현해주고 싶었던거죠? 혹은 새로운 task가 추가되지 않는다는 걸 더 드러내는건 어떨까요?

hannut91 commented on src/reducer.test.jsx:
```diff
@@ -0,0 +1,85 @@
+      });
+    });
+
+    context('without task title', () => {
```
`task title이 없이` 라기 보다는 빈 문자열인 것 같아요. `with empty text`는 어떨까요?

hannut91 commented on src/reducer.test.jsx:
```diff
@@ -0,0 +1,85 @@
+
+        expect(state.taskTitle).toBe('');
+      });
+    });
```
`context`안에 Given을 명시해서 어떤 것이 변하는지 더 명확하게 해주는 것도 좋은 것 같아요
```js
context('with task title', () => {
  const taskTitle = '첫 번째 할 일';

  it('appends task into tasks', () => {'
    const state = reducer(reduceAddTask(taskTitle), action);
  });
});

context('with empty text', () => {
  const taskTitle = '';

  it('appends nothing', () => {
    const state = reducer(reduceAddTask(taskTitle), action);
  });
});
```

bbhye1 committed:
App test을 리팩토링했다.

mock 함수들을 생성해서 test할 수 있도록 만들었다.

bbhye1 committed:
App test를 리팩토링한다.

mock함수들을 이용해서 App을 test할 수 있도록 만든다.

bbhye1 committed:
App.jsx을 리팩토링한다.

action들을 분리시켜주었다.

bbhye1 committed:
reducer test를 리팩토링한다.

action들을 분리시켜서 사용했다.

hannut91 commented on src/App.test.jsx:
```diff
@@ -2,15 +2,29 @@ import React from 'react';
   expect(getByText(/추가/)).not.toBeNull();
 
+  expect(getByText(/첫번째 할 일/)).not.toBeNull();
+  expect(getByText(/두번째 할 일/)).not.toBeNull();
```
`첫 번째 할 일`과 `두 번째 할 일은 Magic literal인 것 같아요.
`tasks`로 표현해줄 수 있을까요? 스터디 때 배운 Side effect를 일으키고, 모든 항목을 순회한다는 의도를 드러낼 수 있는 `forEach`를 쓰면 좋을 것 같네요

See also
  - https://refactoring.com/catalog/replaceMagicLiteral.html

bbhye1 replied:
이부분은 앞으로 항상 잘 염두하겠습니다!

hannut91 commented on src/actions.js:
```diff
@@ -0,0 +1,19 @@
+    type: 'updateTaskTitle',
+    payload: { title: taskTitle },
+  };
+}
```
`taskTitle`대신에 `title`로 사용하는건 어떨까요? 그러면 좀 더 단순해질 것 같아요

```js
export function updateTaskTitle(title) {
  return {
    type: 'updateTaskTitle',
    payload: { title },
  };
}
```

그런데 밑에서보니 또 title에서 taskTitle로 저장하네요. 일관성있게 통일시켜주는게 좋을 것 같아요

hannut91 commented on src/reducer.test.jsx:
```diff
@@ -0,0 +1,67 @@
+  });
+
+  describe('deleteTask', () => {
+    it('remove task from tasks', () => {
+      const state = reducer({
+        tasks: [
+          { id: 1, title: '첫번째 할 일' },
+          { id: 2, title: '두번째 할 일' },
+        ],
+      }, deleteTask(1));
+
+      expect(state.tasks).not.toContain({ id: 1 });
+      expect(state.tasks).toHaveLength(1);
+    });
```
계속 1이라는게 반복되고 있어요. 이정도 간단한 테스트코드에서는 사실 다 자세히 보면 파악이 가능하긴한데, 복잡한 테스트의 경우 왜 1인지 2가 아니면 안되는지 파악하기 어려울 것 같아요.


```js
describe('deleteTask', () => {
  const id = 1;
  const tasks = [
    { id, title: '첫번째 할 일' },
  ];

  it('remove task from tasks', () => {
    const state = reducer({ tasks }, deleteTask(id));

    expect(stats.tasks).toBeEmpty();
  });
});
```

음.. 이건 너무 복잡해보일까요? `tasks`의 `id`가 `id`라는 걸 파악하기 어려울까요? 아니면

```js
describe('deleteTask', () => {
  const tasks = [
    { id: 1, title: '첫번째 할 일' },
  ];

  it('remove task from tasks', () => {
    const state = reducer({ tasks }, deleteTask(tasks[0].id));

    expect(stats.tasks).toBeEmpty();
  });
});
```

이건 어떨까요

bbhye1 replied:
마지막   expect(state.tasks).toHaveLength(1); 이 부분만 남기고 수정해 주었습니다. 왠지 만약 하나 눌렀을 때 모두 지워지는 문제가 생겨서 찾아야 한다면 이 부분이 필요할 것같아서요 ㅎㅎ

bbhye1 committed:
reducer test를 전체적으로 리팩토링한다.

given을 더 명시적일 수 있게 바꾸고 it과 context설명을 더 명확한 표현으로
변경한다.

bbhye1 committed:
actions.js의 파라미터명을 변경한다.

updateTaskTitle의 함수 파라미터를 taskTitle에서 title로 변경하여
payload의 title과 통일시켜주었다.

bbhye1 committed:
App test 리팩토링하기

App test에서 각각 개별로 tasks가 목록으로 잘 보이는지를 forEach메서드를
이용하여 테스트하는걸로 변경했다.

bbhye1 committed:
reducerr에 initialState 테스트를 추가했다.

아무런 action도 없을 때 초기값이 세팅되는 것을 테스트했다.

bbhye1 committed:
reducer initialState를 구현했다.

action에도 초기값을 주어서 구현완료했다.

bbhye1 committed:
ListContainer의 테스트를 새로 작성한다.

bbhye1 committed:
ListContainer을 구현한다.

bbhye1 committed:
Page에서 List를 ListContainer로 대체한다.

bbhye1 committed:
Page test 오류를 수정한다.

mock함수를 이용해 오류를 수정한다.

bbhye1 committed:
App.jsx에서 불필요한 부분을 제거한다.

bbhye1 committed:
InputCotainer에 새 테스트를 작성한다.

bbhye1 committed:
InputContainer를 구현했다.

bbhye1 committed:
InputContainer에 테스트를 추가한다.

dispatch에 대한 테스트를 더 추가한다.

bbhye1 committed:
Page와 Page의 테스트를 리팩토링한다.

필요하지 않은 부분을 제거해준다.

bbhye1 committed:
App에서 필요없는 부분을 모두 제거해준다.

필요한 부분만 남기고 모두 제거해준다.

bbhye1 committed:
Page와 Page 테스트를 제거한다.

App컴포넌트와 Page컴포넌트ㅑ가 거의 동일해져서 Page컴포넌트를 지우고
App컴포넌트에 통합한다.Page와 Page 테스트를 제거한다.

bbhye1 committed:
ListContainer dispatch 테스트를 추가한다.

ListContainer test에 dispatch관련 test내용을 추가한다.

bbhye1 committed:
store.js에 lint오류를 해결한다.

세미콜론 삽입으로 lint오류를 해결한다.

bbhye1 committed:
ListContainer의 테스트에 actions를 적용시킨다.

actions의 action을 테스트에서 사용한다.

hannut91 commented on src/App.test.jsx:
```diff
@@ -2,15 +2,31 @@ import React from 'react';
 
-  // TODO: 통합 테스트 코드 작성
-  // CodeceptJS => 실제 브라우저에서 사용자 테스트 실행 가능.
+  tasks.forEach((task) => {
+    expect(getByText(task.title)).not.toBeNull();
+  });
```
```js
const { container, getByText } = render((
  <App />
));

tasks.forEach(({ title }) => {
  expect(container).toHaveTextContent(title);
});
```

텍스트가 있는지 확인할 때는 간단하게 이렇게 할 수도 있을 것 같아요.

See also
  - https://github.com/testing-library/jest-dom#tohavetextcontent

hannut91 commented on src/InputContainer.test.jsx:
```diff
@@ -0,0 +1,38 @@
+  const taskTitle = '';
+  const newTaskTitle = 'New Title';
+
+  const dispatch = jest.fn();
+
+  useSelector.mockImplementation((selector) => selector({
+    taskTitle,
+  }));
+
+  useDispatch.mockImplementation(() => dispatch);
```
`const dispatch`변수를 선언한 곳과 사용하는과 거리가 먼 것 같아요. 관련이 있는 것들끼리는 가까이에 같이 모여있는건 어떨까요?

hannut91 commented on src/InputContainer.test.jsx:
```diff
@@ -0,0 +1,38 @@
+  expect(getByText(/추가/)).not.toBeNull();
+  expect(getByDisplayValue(taskTitle)).not.toBeNull();
+
+  fireEvent.click(getByText(/추가/));
+
+  expect(dispatch).toBeCalledWith(addTask());
+
+  fireEvent.change(getByDisplayValue(taskTitle), { target: { value: newTaskTitle } });
+
+  expect(dispatch).toBeCalledWith(updateTaskTitle(newTaskTitle));
```
두 가지 테스트가 하나의 테스트 묶음에 같이 있는 것 같아요. 하나의 테스트는 하나의 역할만 하도록 하는건 어떨까요?

hannut91 commented on src/InputContainer.test.jsx:
```diff
@@ -0,0 +1,38 @@
+  ));
+
+  expect(getByText(/추가/)).not.toBeNull();
+  expect(getByDisplayValue(taskTitle)).not.toBeNull();
```
`getByLabelText`를 이용해서 `input`을 가져오는 건 어떨까요? 그러면 taskTitle을 사용하지 않아도 되고, newTaskTitle도 이름을 taskTitle이나 다른 이름을 사용할 수 있을거에요.

hannut91 commented on src/ListContainer.jsx:
```diff
@@ -0,0 +1,28 @@
+  return (
+    <List
+      tasks={tasks}
+      onClickDelete={handleClickDeleteTask}
```
`onClickComplete`는 어떨까요? 만약 삭제하는 대신, 완료 목록에 넣기로 한다면 이름이 조금 모호할 수 있을 것 같아요

hannut91 commented on src/reducer.js:
```diff
@@ -0,0 +1,42 @@
+};
+
+function reducer(state = initialState, action = { type: 'initialState' }) {
+  if (action.type === 'updateTaskTitle') {
+    const { title } = action.payload;
+
+    return {
+      ...state,
+      taskTitle: title,
+    };
+  }
+
+  if (action.type === 'addTask') {
+    const { newId, taskTitle, tasks } = state;
+
+    if (!taskTitle) return state;
+
+    return {
+      ...state,
+      newId: newId + 1,
+      taskTitle: '',
+      tasks: [...tasks, { id: newId, title: taskTitle }],
+    };
+  }
+
+  if (action.type === 'deleteTask') {
+    const { tasks } = state;
+    const { id } = action.payload;
+
+    return {
+      ...state,
+      tasks: tasks.filter((task) => task.id !== id),
+    };
+  }
+  return state;
```
`if`나 switch문 없이 이걸 구현할 수 있을까요?

hannut91 commented on src/reducer.test.jsx:
```diff
@@ -0,0 +1,88 @@
+} from './actions';
+
+describe('reducer', () => {
+  describe('initialState', () => {
+    it('set the inital state', () => {
+      const state = reducer();
+
+      expect(state.taskTitle).toBe('');
+      expect(state.tasks).toHaveLength(0);
+      expect(state.newId).toBe(100);
+    });
+  });
```
정의되어 있지 않은 action이 주어졌을 때를 테스트하고 싶으셨던 거죠/

hannut91 commented on src/reducer.test.jsx:
```diff
@@ -0,0 +1,88 @@
+        taskTitle,
+      }, updateTaskTitle(newTaskTitle));
+
+      expect(state.taskTitle).toBe('New title');
```
여기도 Magic Literal이 사용됐네요

hannut91 commented on src/reducer.test.jsx:
```diff
@@ -0,0 +1,88 @@
+  });
+
+  describe('addTask', () => {
+    function reduceAddTask(taskTitle) {
+      return {
+        newId: 100,
+        taskTitle,
+        tasks: [],
+      };
+    }
```
Arrow 함수를 사용하면 어떤 장점이 있었다고 했었죠?
그리고 얘가 반환하는 것이 `state`인거죠?

bbhye1 replied:
음;;;; 이 부분에서 Arrow함수를 왜 언급하신지 이해를 못하겠어요. 얘가 반환하는 것은 state가 맞는데 강의처럼 reducer를 함수안에 넣어주는게 맞는것 같아 일단 그렇게 수정해주었습니다!

hannut91 commented on src/reducer.test.jsx:
```diff
@@ -0,0 +1,88 @@
+      };
+    }
+
+    context('with task title', () => {
+      it('appends task into tasks', () => {
+        const taskTitle = '첫번째 할 일';
+
+        const state = reducer(reduceAddTask(taskTitle), addTask());
+
+        expect(state.tasks).toHaveLength(1);
+        expect(state.tasks[0].title).toBe('첫번째 할 일');
+        expect(state.tasks[0].id).not.toBeUndefined();
+      });
+
+      it('clear task title', () => {
+        const taskTitle = '첫번째 할 일';
+
+        const state = reducer(reduceAddTask(taskTitle), addTask());
+
+        expect(state.taskTitle).toBe('');
+      });
+    });
```
`const taskTitle = '첫번째 할 일';`이게 중복되고 있고, 
이 변수가 테스트에 큰 영향을 주지 않는다면 테스트를 하는 곳 바깥으로 빼도 괜찮을 것 같아요.

무언가가 반복된다면 리팩터링을 해야하는 신호로 보시면 좋을 것 같아요

```js
context('with task title', () => {
  const taskTitle = '첫번째 할 일';
  
  it('appends task into tasks', () => {
  });
});

```

hannut91 commented on src/reducer.test.jsx:
```diff
@@ -0,0 +1,88 @@
+        const state = reducer(reduceAddTask(taskTitle), addTask());
+
+        expect(state.tasks).toHaveLength(1);
+        expect(state.tasks[0].title).toBe('첫번째 할 일');
```
여기도 Magic literal대신에 사용한 변수를 사용하는게 좋을 것 같아요

hannut91 commented on src/reducer.test.jsx:
```diff
@@ -0,0 +1,88 @@
+      }, deleteTask(tasks[0].id));
+
+      expect(state.tasks).not.toContain({ id: tasks[0].id });
+      expect(state.tasks).toHaveLength(1);
```
`expect(state.tasks).toHaveLength(tasks.length - 1);`

이건 어떨까요? 어떤 장점이 있을까요?

bbhye1 replied:
확실히 하나가 제거된다는걸 명시적으로 표현해주네요!!

bbhye1 committed:
App test에서 query와 matcher를 변경한다.

query를 container로 matcher를 toHaveTextContent로 변경한다.

bbhye1 committed:
InputContainer test를 리팩토링한다.

하나의 테스에 하나의 역할을 할 수 있도록 테스트를 나눠주었다.
그리고 getByDisplayValue 대신 getLabelText로 변경했다.

bbhye1 committed:
List에 테스트를 수정한다.

bbhye1 committed:
List와 ListContainer에서 빠르게 구현한다.

bbhye1 committed:
ListContainer에서 함수명을 변경한다.

bbhye1 committed:
Item에 테스트를 수정한다.

bbhye1 committed:
List와 Item을 수정해서 빠르게 테스트를 통과한다.

bbhye1 committed:
reducer 테스트에서 Magic literal를 제거한다.

bbhye1 committed:
reducer 테스트를 수정한다.

reducer 파라미터에 state 기본값을 설정해준다.

bbhye1 committed:
reducer를 리팩토링한다: actions에 updateTaskTitle 추가

bbhye1 committed:
reducer를 리팩토링한다: actions에 addTask 추가

bbhye1 committed:
reducer를 리팩토링한다: actions에 deleteTask 추가

bbhye1 committed:
reducer를 리팩토링한다. :if문을 삭제.

bbhye1 committed:
reducer에서 중복을 제거한다.

bbhye1 committed:
reducer 테스트를 수정한다.

reducer에서 숫자 1 대신 tasks.length - 1 로 표현하여 의도를 드러내준다.

bbhye1 committed:
reducer의 테스트 메세지를 변경한다.

bbhye1 committed:
reducer에서 중복을 제거해주었다.

bbhye1 committed:
reducer의 테스트에서 Magic Literal를 제거한다.

bbhye1 committed:
InputContainer의 테스트이름을 변경한다.

bbhye1 committed:
Input 테스트를 나누어 역할을 나눠준다.

bbhye1 committed:
ListContainer에서 줄 위치를 변경한다.

hannut91 commented on src/App.test.jsx:
```diff
@@ -2,15 +2,31 @@ import React from 'react';
 
-  // TODO: 통합 테스트 코드 작성
-  // CodeceptJS => 실제 브라우저에서 사용자 테스트 실행 가능.
+  tasks.forEach((task) => {
+    expect(container).toHaveTextContent(task.title);
+  });
```
구조 분해 할당을 이용해서 이렇게 표현하는 건 어떨까요?

```suggestion
  tasks.forEach(({ title }) => {
    expect(container).toHaveTextContent(title);
  });
```

hannut91 commented on src/Input.test.jsx:
```diff
@@ -4,27 +4,37 @@ import { render, fireEvent } from '@testing-library/react';
 
-test('Input', () => {
+describe('Input', () => {
   const handleChange = jest.fn();
   const handleClick = jest.fn();
```
mocking함수는 초기화를 해주지 않으면 의도치 않게 동작할 수 있습니다. 예를들어 테스트 A, B가 있는데, A에서 `handleChange`를 실행해버리면 실행된 상태가 그대로 남아 있기 때문에 B에서 `handleChange`실행하지 않았더라도 확인해보면 실행되었다고 결과가 출력됩니다. 따라서 항상 초기화를 해주는게 좋습니다.

```js
beforeEach(() => {
  jest.clearAllMocks();
});
```

See also
  - https://jestjs.io/docs/en/setup-teardown
  - https://jestjs.io/docs/en/jest-object

hannut91 commented on src/Input.test.jsx:
```diff
@@ -4,27 +4,37 @@ import { render, fireEvent } from '@testing-library/react';
   });
 
-  expect(handleChange).toBeCalled();
+  it('renders "추가" button', () => {
```
`추가`라는 말은 구체적인 구현에 의존하기 때문에 더 추상적인 이름이 좋을 것 같아요. 
만약 추가 대신에 `+`아이콘을 사용하기만 해도 이 테스트의 이름은 다시 수정해야 될 것 같아요

hannut91 commented on src/InputContainer.test.jsx:
```diff
@@ -0,0 +1,47 @@
+  }));
+
+  const dispatch = jest.fn();
+  useDispatch.mockImplementation(() => dispatch);
```
초기화하는 로직은 여기에 바로 쓰기 보다는 `beforeEach`안에서 작성하는게 좋을 것 같아요. 왜냐하면 테스트를 실행할 때 마다 항상 새로운 상태에서 시작해야되기 때문이에요. 이러면 테스트가 다른 테스트에 영향을 주게 됩니다. 그러면 테스트를 유지보수하기 굉장히 어려워질거에요. 철저하게 분리해야 합니다.

hannut91 commented on src/InputContainer.test.jsx:
```diff
@@ -0,0 +1,47 @@
+  const dispatch = jest.fn();
+  useDispatch.mockImplementation(() => dispatch);
+
+  it('dispatch updateTaskTitle action', () => {
```
테스트대상이 무엇인지 모르겠어요. Input 컨테이너가 `updateTaskTitle` 액션을 dispatch한다는 일만 명시되어있고 실제로 무엇을 테스트하는건지 잘 모르겠어요.

```js
describe('InputContainer', () => {
  describe('change taskTitle', () => {
    it('dispatches updateTaskTitle action', () => {
      ...
    });
  });
});
```

hannut91 commented on src/reducer.js:
```diff
@@ -0,0 +1,25 @@
+      tasks: state.tasks.filter((task) => task.id !== id),
+    },
+  };
+  return { ...state, ...actions[action.type] };
```
상태에 액션이 들어가있는 것 같아요. 그리고 액션이 아니라 상태를 변경하는 함수는 `reducer`라고 불러요.

bbhye1 committed:
App의 테스트에 구조분해할당을 적용한다.

bbhye1 committed:
Input test를 리팩토링한다.

구조분해할당으로 간결하게 표현하고 beforeEach로 함수초기화를 시켰다.
그리고 테스트명을 구현에 의존하지 않을 수 있도록 변경한다.

bbhye1 committed:
InputContainer와 ListContainer의 테스트에 beforeEach를 적용한다.

bbhye1 committed:
reducer를 리팩토링한다.

bbhye1 committed:
List 테스트에서 테스트명을 변경한다.

구현에 의존하지 않도록 테스트 이름을 "완료"에서 complete로 변경했다.

bbhye1 committed:
InputContainer에서 태스트 대상을 구체화한다.

hannut91 commented on src/reducer.js:
```diff
@@ -4,22 +4,31 @@ const initialState = {
+};
+
+function reducer(state = initialState, action = { type: 'initialState' }) {
+  if (!reducers[action.type]) return state;
```
한 줄 이더라도 중괄호를 써주는게 더 좋을 것 같아요. 팀의 컨벤션이나 eslint rule에 따라 다르긴 하지만, 한 줄 이더라도 중괄호를 넣자는 파와 그냥 간단하게 쓰자는 파로 나뉩니다.
저는 무조건 써주는걸 선호하긴합니다.

See also
  - https://eslint.org/docs/rules/curly

hannut91 reviewed (approved):
다 좋습니다 ㅎㅎ

bbhye1 committed:
reducer의 if문을 수정한다.
